sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -706,6 +706,28 @@ object functions {
     }
   }
 
+  /**
+   * Aggregate function: returns the product of all numerical elements in a group.
+   *
+   * @param scale   A scaling constant to be multiplied into each non-null value
+   *                in the product, in order to provide some defence against overflow.
+   *                By default, this is set to one.
+   *
+   * @group agg_funcs
+   * @since 3.2.0
+   */
+  def product(e: Column, scale: Double = 1.0): Column =
```

Comment:
`scale: Double = 1.0` -> let's don't use default value. It doesn't work in Java.

Reply:
Thanks @HyukjinKwon for this and your other helpful comments. I'll deal with these shortly.